homeassistant/components/matrix/config_flow.py

```diff
@@ -0,0 +1,67 @@
+"""Config flow to configure Matrix Bot."""
+import functools
+import logging
+
+from matrix_client.client import MatrixClient, MatrixRequestError
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, CONF_VERIFY_SSL
+
+from .const import CONF_HOMESERVER, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    client = MatrixClient(
+        base_url=data[CONF_HOMESERVER], valid_cert_check=data[CONF_VERIFY_SSL],
+    )
+    try:
+        await hass.async_add_executor_job(
+            functools.partial(client.login, data[CONF_USERNAME], data[CONF_PASSWORD],)
+        )
+    except MatrixRequestError:
+        raise InvalidAuth
+
+    return {"title": "MatrixServer"}
```

Comment:
Can we get the name of the server? And should we include the username in the title?